lib/litani_report.py

```diff
@@ -547,9 +547,13 @@ def render(run, report_dir):
 
     pipe_templ = env.get_template("pipeline.jinja.html")
     for pipe in run["pipelines"]:
-        PipelineDepgraphRenderer.render(
-            render_root=temporary_report_dir,
-            pipe_url=pathlib.Path(pipe["url"]), pipe=pipe)
+        if shutil.which("dot"):
```

Comment:
This is still pretty inefficient. The problem is that `litani_report.render()` gets called repeatedly during the run, and within `litani_report.render()` this for loop iterates over every pipeline. This code is calling `which` on every single iteration, there's no need for that... if `dot` wasn't installed five seconds ago it's not going to be installed now either :)

I was hoping something more like: when you construct a `PipelineDepgraphRenderer`, it calls `which` just once to find if dot is installed. If it is not installed, then `PipelineDepgraphRenderer.render()` should become a no-op, so it doesn't matter that we call it repeatedly.